### PR TITLE
Add script detection to script alias

### PR DESCRIPTION
**Pull request by PrinsFrank**

### CODE REVIEW

**Comment by szepeviktor** [comment]
Why not test a string for only one script?
Aren't ISO script names in one script only?

**Comment by PrinsFrank** [comment]
> Why not test a string for only one script? Aren't ISO script names in one script only?

I tried to make this as generic as possible as these new methods are in the src folder - not in the dev folder - so anyone can use them.

**Comment by szepeviktor** [comment]
Will it be able to tell whether a string is in a single script?

File: tests/Unit/Scripts/ScriptAliasTest.php [hunk 1]
```diff
@@ -45,4 +45,53 @@ public function testIsSupportedByPHPRegex(): void
...
+    /** @covers ::allForString */
+    public function testForString(): void
+    {
+        static::assertSame(
+            [],
+            ScriptAlias::allForString('')
+        );
+        static::assertSame(
+            [
+                ScriptAlias::Latin
```
Comment (szepeviktor) [inline tests/Unit/Scripts/ScriptAliasTest.php:58]:
```suggestion
                ScriptAlias::Latin,
```

**Comment by PrinsFrank** [comment]
> Will it be able to tell whether a string is in a **single** script?

It can be done with this line of code `count(ScriptAlias::allForString($string)) === 1`. Do you have any suggestions? I'm also not very happy with how complex the method turned out, any suggestion to make it simpler?

**Comment by szepeviktor** [comment]
> I'm also not very happy with how complex the method turned out, any suggestion to make it simpler?

No.

File: src/Scripts/ScriptAlias.php [hunk 2]
```diff
@@ -401,4 +401,37 @@ public function isSupportedByPHPRegex(): bool
     {
         return BackedEnum::hasCaseAttribute($this, SupportedByPHPRegex::class);
     }
+
+    /**
+     * @return $string is non-empty-string ? non-empty-array<self> : [] in order of most matched multibyte characters
+     *
+     * Please note that not all Scripts are supported, only the ones that have the 'SupportedByPHPRegex' attribute.
+     * For all other scripts, self::Command will be returned
```
Comment (szepeviktor) [inline src/Scripts/ScriptAlias.php:409]:
```suggestion
     * For all other scripts, self::Common will be returned
```

Comment (PrinsFrank) [reply src/Scripts/ScriptAlias.php:409]: Whoops, thanks!

**Comment by szepeviktor** [comment]
I'm looking forward to see this in action! `eu_greek`

File: src/Scripts/ScriptAlias.php [hunk 3]
```diff
@@ -401,4 +401,42 @@ public function isSupportedByPHPRegex(): bool
     {
         return BackedEnum::hasCaseAttribute($this, SupportedByPHPRegex::class);
     }
+
+    /**
+     * @return ($string is non-empty-string ? non-empty-array<ScriptAlias> : array) in order of most matched multibyte characters
```
Comment (szepeviktor) [inline src/Scripts/ScriptAlias.php:406]:
```suggestion
     * @return ($string is non-empty-string ? non-empty-array<ScriptAlias> : empty-array) in order of most matched multibyte characters
```

**Comment by szepeviktor** [comment]
Sorry, there is no such a type as `empty-array`, it is `array{}`

**Comment by szepeviktor** [comment]
GHA can cancel previous workflow runs: https://github.com/szepeviktor/byte-level-care/blob/70bb8404654b91e97c20f982c70f7d0ca0ebb98c/.github/workflows/back-end.yml#L17-L19

**Pull request merged**